patroni/daemon.py

```diff
@@ -8,16 +8,299 @@
+        val = parsed_dsn.get(conn_param, os.getenv(env_var))
+        if val:
+            su_params[conn_param] = val
+    # because we use "username" in the config for some reason
+    su_params['username'] = su_params.pop('user', getuser())
+    su_params['password'] = su_params.get('password') or getpass('Please enter the user password:')
+
+    from . import psycopg
+    try:
+        conn = psycopg.connect(dsn=dsn, password=su_params['password'])
+    except psycopg.Error as e:
+        print(f'Failed to establish PostgreSQL connection: {e}', file=sys.stderr)
+        sys.exit(1)
+
+    with conn.cursor() as cur:
+        cur.execute("SELECT 1 FROM pg_roles WHERE rolname=%s AND rolsuper='t';", (su_params['username'],))
+        if cur.rowcount < 1:
+            print('The provided user does not have superuser privilege', file=sys.stderr)
+            sys.exit(1)
+
+        cur.execute("SELECT name, current_setting(name) FROM pg_settings \
+                     WHERE context <> 'internal' \
+                     AND source IN ('configuration file', 'command line', 'environment variable') \
+                     AND category <> 'Write-Ahead Log / Recovery Target' \
+                     AND setting <> '(disabled)' \
+                     OR name IN ('hba_file', 'ident_file', 'config_file', \
+                                 'data_directory', \
+                                 'listen_addresses', 'port', \
+                                 'max_connections', 'max_worker_processes', 'max_wal_senders', \
+                                 'max_replication_slots', 'max_locks_per_transaction', 'max_prepared_transactions', \
+                                 'hot_standby', \
+                                 'wal_level', 'wal_log_hints', \
+                                 'wal_keep_segments', 'wal_keep_size', \
+                                 'track_commit_timestamp');")
+
+        # adjust values
+        for p, v in cur.fetchall():
+            if p == 'data_directory':
+                config['postgresql']['data_dir'] = v
+            elif p == 'cluster_name':
+                config['scope'] = v
+            elif p in ('archive_command', 'restore_command', 'archive_cleanup_command',
+                       'recovery_end_command', 'ssl_passphrase_command',
+                       'hba_file', 'ident_file', 'config_file'):
+                # write commands to the local config due to security implications
+                # write hba/ident/config_file to local config to ensure they are not removed later
+                config['postgresql'].setdefault('parameters', {})
+                config['postgresql']['parameters'][p] = v
+            else:
+                config['bootstrap']['dcs']['postgresql']['parameters'][p] = v
+
+    conn.close()
+
+    port = config['bootstrap']['dcs']['postgresql']['parameters']['port']
+    connect_host = parsed_dsn.get('host', os.getenv('PGHOST')) or get_local_ip()
```

Comment:
When executed locally the `host` or `PGHOST` most likely will be not set at all or set to something like `localhost` or `/tmp`.
The last one will create the problem.
```suggestion
    connect_host = get_local_ip()
```